src/coreclr/jit/lsraxarch.cpp

```diff
@@ -2882,7 +2883,14 @@ void LinearScan::SetContainsAVXFlags(unsigned sizeOfSIMDVector /* = 0*/)
         compiler->GetEmitter()->SetContainsAVX(true);
         if (sizeOfSIMDVector == 32)
         {
-            compiler->GetEmitter()->SetContains256bitAVX(true);
+            compiler->GetEmitter()->SetContains256bitOrMoreAVX(true);
+        }
+    }
+    if (compiler->canUseEvexEncoding())
```

Comment:
This can be nested in the `canUseVexEncoding` path since it can never be `true` if `canUseVexEncoding` is false.

You might also consider inverting the condition to reduce nesting. That is:
```
if (!compiler->canUseVexEncoding())
{
    return;
}

// ...

```

Reply:
```
void LinearScan::SetContainsAVXFlags(unsigned sizeOfSIMDVector /* = 0*/)
{
    if (!compiler->canUseVexEncoding())
    {
        return;
    }

    compiler->compExactlyDependsOn(InstructionSet_AVX);
    compiler->GetEmitter()->SetContainsAVX(true);
    if (sizeOfSIMDVector == 32)
    {
        compiler->GetEmitter()->SetContains256bitOrMoreAVX(true);
        return;
    }

    if (compiler->canUseEvexEncoding())
    {
        if (compiler->compExactlyDependsOn(InstructionSet_AVX512F) && (sizeOfSIMDVector == 64))
        {
            compiler->GetEmitter()->SetContains256bitOrMoreAVX(true);
        }
    }
}
```